src/twisted/conch/ssh/keys.py

```diff
@@ -1241,8 +1402,20 @@ def verify(self, signature, data):
                 k = k.public_key()
             verifier = k.verifier(
                 signature, hashes.SHA1())
-        else:
-            raise BadKeyError("unknown key type %s" % (self.type(),))
+
+        elif keyType == 'EC':
+            k = self._keyObject
+            if not self.isPublic():
+                k=k.public_key()
```

Comment:
Sorry I missed this earlier. I addressed this comment and also added tests in `signAndVerify` using private RSA and DSA keys, even though it is not related to this ticket but helps with the project coverage.